### PR TITLE
External dns fixes

### PR DESCRIPTION
**Pull request by olemarkus**

### CODE REVIEW

File: docs/cluster_spec.md [hunk 1]
```diff
@@ -585,9 +585,11 @@ You can use [external-dns](https://github.com/kubernetes-sigs/external-dns/) as
 ```yaml
 spec:
   externalDns:
-    provider: ExternalDNS
+    provider: external-dns
 ```
 
+Note that you if you have dns-controller installed, you need to remove this deployment before updating the cluster with the new configuration.
```
Comment (rifelpet) [inline docs/cluster_spec.md:591]: can we use our pruning support to delete it instead?

**Review by rifelpet — APPROVED** [review]

File: docs/cluster_spec.md (hunk 1 above)
Comment (olemarkus) [reply docs/cluster_spec.md:591]: When/if it merges :)

**Review by rifelpet — APPROVED** [review]

**Pull request merged**